_posts/2020-11-03-proces-migracji-do-chmury.md

```diff
@@ -0,0 +1,233 @@
+---
+layout:    post
+title:     "Proces migracji do chmury - czyli od czego zacząć?"
+date:      2020-11-03 08:00:00 +0100
+published: true
+author:    mhoja
+image:     /assets/img/posts/2020-11-03-proces-migracji-do-chmury/clouds.jpg
+tags:
+    - cloud
+    - serverless
+    - gcp
+    - aws
+    - azure
+    - googlecloud
+---
+
+Planując migrację systemu informatycznego do chmury, zastanawiamy się *od czego zacząć*? Proces migracji składa się z kilku etapów, które pomogą nam przejść z punktu A (systemu źródłowego), do punktu B (systemu docelowego).
+
+Przed rozpoczęciem lektury warto zapoznać się z wpisem Grzegorza pt.  
+["W chmurze czyli jak? O możliwych kierunkach rozwoju aplikacji chmurowych"]({% post_url 2020-10-22-modele-wdrozenia-w-chmurze %})  
+który przedstawia poszczególne modele wdrożenia w chmurze.
+
+Swoją wiedzę opierać będę na doświadczeniach z chmurą Google ([Google Cloud Platform, GCP](https://cloud.google.com/)), dlatego w tekście będę nawiązywał do usług i pojęć związanych z tą platformą.
+
+![Etapy migracji do chmury](/assets/img/posts/2020-11-03-proces-migracji-do-chmury/etapy_migracji.jpg)
+
+# Szacowanie
+
+Pierwszym z etapów migracji jest etap szacowania, na którym przeprowadzona zostaje dokładna analiza istniejącego systemu oraz infrastruktury.
+Podczas szacowania analizujemy wykorzystywane zasoby, zależności pomiędzy elementami systemu oraz wymagania.
+Wybieramy te elementy systemu, które chcemy przenieść do chmury, co pozwoli na obliczenie dokładnych kosztów utrzymania infrastruktury.
+W przypadku kiedy przenosimy maszyny wirtualne, szacujemy zasoby które są nam potrzebne, aby określić np. typ maszyn [Compute Engine](https://cloud.google.com/compute) oraz ich parametry.
+
+Jeśli znamy już dokładnie system źródłowy i jego wymagania, musimy zapoznać się z platformą usługodawcy, do którego będziemy migrować nasz system. Próg wejścia będzie tutaj zależał od modelu wdrożenia, ponieważ w zależności od niego musimy poznać mniej lub bardziej całą platformę.
+
+Pierwszy etap migracji jest tym miejscem, w którym zapoznajemy się z możliwościami chmury oraz zdobywamy potrzebne do migracji kompetencje (na przykład poprzez szkolenia czy certyfikacje).
+Pomóc może nam w tym również przetestowanie platformy, wdrożenie PoC oraz eksperymenty, które pozwolą rozwiać wszelkie wątpliwości i przewidzieć potencjalne problemy.
+
+**Rozważyć należy między innymi:**
+
+- porównanie wydajności i możliwości aktualnej bazy danych z usługami bazodanowymi w chmurze (np. [Cloud SQL, Cloud Firestore](https://cloud.google.com/products/databases)),
```

Comment:
to jeszcze dorzuć killera od gcp: https://cloud.google.com/spanner
;-)

Reply:
Dorzucam, niech będzie :) Chciałem dodać tylko jeden przykład bazy relacyjnej i nierelacyjnej, żeby za chwilę nie wypisywać wszystkich, ale specjalnie dla Ciebie Spanner również się znajdzie.

Reply:
to może lepiej wywalić cloud sql i dać spannera. jak ktoś przeczyta jak działa cloud sql no to pomyśli 'co to mi w sumie daje'. a przecież firestore to state of the art ;) spanner to też już wariant 'big f cloud', state of the art ;-)

Reply:
zostawię 3 przykłady, bo nie powoduje to rozjechania tekstu/przeskoczenia do nowej linii - jest ok :)